sdk/appconfiguration/azure-data-appconfiguration/src/samples/java/com/azure/data/appconfiguration/WatchFeatureAsync.java

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.appconfiguration;
+
+import com.azure.data.appconfiguration.models.ConfigurationSetting;
+import com.azure.data.appconfiguration.models.SettingSelector;
+import reactor.core.publisher.Flux;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class WatchFeatureAsync {
+    /**
+     * Runs the sample algorithm and demonstrates how to read configuration setting revision history.
+     *
+     * @param args Unused. Arguments to the program.
+     */
+    public static void main(String[] args) {
+        // The connection string value can be obtained by going to your App Configuration instance in the Azure portal
+        // and navigating to "Access Keys" page under the "Settings" section.
+        String connectionString = "endpoint={endpoint_value};id={id_value};secret={secret_value}";
+
+        // Instantiate a client that will be used to call the service.
+        final ConfigurationAsyncClient client = new ConfigurationClientBuilder()
+                                                    .connectionString(connectionString)
+                                                    .buildAsyncClient();
+
+        // Prepare a list of watching settings and update one same setting value to the service.
+        String prodDBConnectionKey = "prodDBConnection";
+        String prodDBConnectionLabel = "prodLabel";
+
+        List<ConfigurationSetting> settings = Flux.concat(
+            client.addConfigurationSetting(prodDBConnectionKey, prodDBConnectionLabel, "prodValue"),
+            client.addConfigurationSetting("stageDBConnection", "stageLabel", "stageValue"))
+            .then(client.listConfigurationSettings(new SettingSelector().setKeyFilter("*")).collectList()).block();
```

Comment:
We shouldn't use blocking in our asynchronous examples, blocking is generally not preferred and can lead to runtime exceptions if it is used in the wrong spot

Reply:
updated